### DISCUSSION TITLE
Why Open WebUI needs GPU

### DISCUSSION THREAD

**Discussion by haidescs** [open-webui/open-webui#2167] [Q&A] [answered]
I see the new-added command as the following:

#To run Open WebUI with Nvidia GPU support, use this command:
#docker run -d -p 3000:8080 --gpus all --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:cuda

This optional command confused me, because based on the introduction open_webui is just a webui of ollama running as server side, so theoretically it doesn't need the GPU. 
#Open WebUI is an extensible, feature-rich, and user-friendly self-hosted WebUI designed to operate entirely offline. It supports various LLM runners, including Ollama and OpenAI-compatible APIs. 

So my question is why there is this command, or when should I use this command to install open_webui?

PLease help.

**Comment by cheahjs** [comment, marked as answer]
Open WebUI has features that can take advantage of GPU acceleration. Right now these include:

1. Running embedding models on documents for RAG
2. Speech-to-text using Whisper models

**Reply by haidescs** [reply]
Thank you very much for your answer. Do you mean that if I misuse this feature to install open_webui and which I want is to use lama, this feature will make open_webui snatches GPU resource with lama? Am I right?

**Comment by justinh-rahb** [comment]
The proper answer _as of now_ is that this is for advanced users, that wish to use larger embedding or STT models. Will we have more uses for GPU support in future? Definitely.

**Reply by haidescs** [reply]
Yes you are right and thank you. When I install the lama, I use open_webui, ollama and lama, so I want to make sure there is no competition of GPU resource between open_webui and lama if I used this command to install open_webui.

**Comment by moodler** [comment]
This may help someone: Right now, Docker does not support use of Metal/GPUs on Apple Silicon.   So you may want to avoid using Docker (for Ollama especially) if you care about speed.
